### DISCUSSION TITLE
Weird commit loading behavior since a few iOS betas

### DISCUSSION THREAD

**Discussion by FelixHerrmann** [community/community#75788] [Mobile] [answered]
### Select Topic Area

Bug

### Body

Hello, I’m experiencing a weird commit loading behavior on the iOS app when an older snapshot of commits is cached and new commits happened since then, it is always necessary to refresh manually to get the latest ones; this happens every time and started behaving like that a few (beta) versions ago.

Here an example:
https://github.com/community/community/assets/42500484/c2e91b75-b8a8-4b9e-b424-240e9782a1cd

**Comment by HenSquared** [comment]
Hi @FelixHerrmann ,

Thanks a lot for the feedback. Sorry about that issue. 

I've opened an issue internally to address this. I believe we know the root cause of this, so I hope we can get something out quickly to fix it. 🤞 

We'll come back to this thread if we have further questions. 

Thanks and happy weekend! :)

**Reply by FelixHerrmann** [reply]
Sounds good, thanks for taking care!

**Reply by HenSquared** [reply, marked as answer]
@FelixHerrmann , We've released an update that should fix this issue. Let me know if you still are facing issues here. 

Thanks for helping make GitHub better 😎

**Reply by FelixHerrmann** [reply]
Hey @HenSquared, thanks for the update. I can confirm that it's fixed; in my initial report I've only mentioned commits but basically every list was affected, all working now as expected!
Thank you and your team 🙏

**Reply by HenSquared** [reply]
Great! Thanks for checking

**Comment by dakshithadissanayaka** [comment]
The issue described by the user seems to be a caching bug in the GitHub iOS app. The app is caching an older snapshot of commits and not automatically refreshing to show the latest changes when new commits are made. This issue started happening a few beta versions ago, suggesting that it might be introduced in a recent update.

Here are some possible solutions to this issue:

1. **Force Refresh:** Manually refresh the commit list by pulling down on the screen. This should force the app to fetch the latest commits from the server.

2. **Clear Cache:** Try clearing the app's cache to remove any outdated or corrupted data. To do this, go to Settings > General > iPhone Storage > Find GitHub > Offload App. This will remove the app's data and reinstall it without deleting your user data.

3. **Update to Latest Beta:** If you're still using an older beta version of the app, try updating to the latest beta version. The latest beta might have fixed the caching bug.

4. **Report the Bug:** If the issue persists, report it to GitHub using their official support channels. Provide detailed information about the bug, including the steps to reproduce it, the device and OS versions, and any relevant screenshots or screen recordings.

In the meantime, you can use the workaround of manually refreshing the commit list to get the latest changes.
